lib/modules/manager/pip-compile/extract.ts

```diff
@@ -206,3 +214,31 @@ export async function extractAllPackageFiles(
   );
   return result;
 }
+
+function extendWithIndirectDeps(
+  packageFileContent: PackageFileContent,
+  lockedDeps: PackageDependency[],
+): void {
+  for (const lockedDep of lockedDeps) {
+    if (
+      !packageFileContent.deps.find(
+        (dep) =>
+          normalizeDepName(lockedDep.depName!) ===
+          normalizeDepName(dep.depName!),
+      )
+    ) {
+      packageFileContent.deps.push(indirectDep(lockedDep));
+    }
+  }
+}
+function indirectDep(dep: PackageDependency): PackageDependency {
+  const result = {
+    ...dep,
+    lockedVersion: dep.currentVersion,
+    depType: 'indirect',
+    enabled: false,
+  };
+  delete result.currentValue;
+  delete result.currentVersion;
```

Comment:
why this? 🤔

Reply:
I'm removing these values, so this treated just like unconstrained dependency. There are some parts that give such dependencies "isLockfileUpdate" flag. So this ensures that. Maybe this object can be recreated in another way?

Reply:
are they set from extraction?

Reply:
Yes, they are parsed by `pip_requirements` manager. I want to keep everything else, so future metadata is not omitted. And just removing these two values makes it behave like any other unconstrained dependency when enabled.

Reply:
Should I add a comment in code why these properties are removed?

Reply:
yes please do that